portal/static/js/gil.js

```diff
@@ -854,214 +955,425 @@ module.exports = OrgTool = (function() {
+      })(__CRSF_TOKEN);
+
+      var readCookie = function(name) {
+        var nameEQ = name + "=";
+        var ca = document.cookie.split(";");
+        for (var i = 0; i < ca.length; i++) {
+            var c = ca[i];
+            while (c.charAt(0) === " ") {
+              c = c.substring(1, c.length);
+            }
+            if (c.indexOf(nameEQ) === 0) {
+              return c.substring(nameEQ.length, c.length);
+            }
+        }
+        return null;
+      };
+
+      // Set default sessionLifetime from Flask config
+      // Subtract 10 seconds to ensure the backend doesn't expire the session first
+      var CONFIG_SESSION_LIFETIME, DEFAULT_SESSION_LIFETIME;
+      var cookieTimeout = readCookie("SS_TIMEOUT");
+      cookieTimeout = cookieTimeout ? parseInt(cookieTimeout) : null;
+
+      if (cookieTimeout && cookieTimeout > 0) {
+        DEFAULT_SESSION_LIFETIME = (cookieTimeout * 1000) - (cookieTimeout > 10 ? (10 * 1000) : 0);
+      } else {
+        try {
+          CONFIG_SESSION_LIFETIME = $("#sessionLifeTime").val();
+          if (!CONFIG_SESSION_LIFETIME || CONFIG_SESSION_LIFETIME === "") {
+            CONFIG_SESSION_LIFETIME = 15 * 60;
+          }
+          DEFAULT_SESSION_LIFETIME = (CONFIG_SESSION_LIFETIME * 1000) - (CONFIG_SESSION_LIFETIME > 10 ? (10 * 1000) : 0);
+        } catch(e) {
+          DEFAULT_SESSION_LIFETIME = (15 * 60 * 1000) - (10 * 1000);
+        }
+      }
+
+      var sessMon = sessionMonitor({
```

Comment:
Could this be moved to a separate file? It seems like something that could be shared between the portal wrapper, GIL and portal.